src/caw/client/go_client/caw_client.go

```diff
@@ -0,0 +1,285 @@
+package caw_client
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/golang/protobuf/ptypes"
+	"github.com/golang/protobuf/ptypes/any"
+
+	faz "github.com/floan/csci499_floan/src/faz/client/go_client"
+
+	caw "github.com/floan/csci499_floan/protos/go_protos/caw"
+)
+
+type CawClient struct {
+	functionToEventType_ map[string]int
+	client_ *faz.FazClient
+}
+
+// This function acts as a constructor 
+// and initializes the member variables 
+// of the CawClient struct
+func CreateCawClient() *CawClient {
+	cawClient := &CawClient{}
+	cawClient.functionToEventType_ = map[string]int{
+		"registeruser": 1,
+		"caw": 2,
+		"read": 3,
+		"follow": 4,
+		"profile": 5,
+	}
+	cawClient.client_ = faz.CreateFazClient("localhost:50000")
+
+	return cawClient
+}
+
+// This function acts as a destructor for 
+// the caw client. All it does is call the 
+// Faz client destructor to close the connection
+func (cawClient *CawClient) DestroyCawClient() {
+	cawClient.client_.DestroyFazClient()
+}
+
+// Registers the user in the kvstore database
+// Args: the users username string
+// Returns: bool to indicate success / failure
+func (cawClient *CawClient) RegisterUser(username string) bool {
+	registerRequest := caw.RegisteruserRequest{
+		Username: username,
+	}
+	anyRequest, errPacking := ptypes.MarshalAny(proto.Message(&registerRequest))
+	if errPacking != nil {
+		fmt.Println(errPacking.Error())
+		return false
+	}
+	var anyResponse = new(any.Any)
+	registerSuccessBool := cawClient.client_.Event(anyRequest, anyResponse, 
+													cawClient.functionToEventType_["registeruser"])
+	// No need to unpack the response for Registeruser
+	if registerSuccessBool {
+		fmt.Println("User registered successfully!")
+		return true
+	} else {
+		errRegistering := fmt.Errorf("There was an error in registering the user")
+		fmt.Println(errRegistering.Error())
+		return false
+	}
+} 
+
+// Creates and saves a caw "post" in the kvstore
+// This also performs the reply functionality
+// when given a parent id
+// Args: posters username and text to post
+// Returns: bool to indicate success / failure
+func (cawClient *CawClient) PostCaw(username string,
+																		text string,
+																		parent_id int) bool {
+	var cawRequest caw.CawRequest
+	// Checking if this is a reply or
+	// a brand new caw
+	if parent_id != -1 {
+		cawRequest = caw.CawRequest{ Username: username, Text: text, 
+																 ParentId: []byte(strconv.Itoa(parent_id)) }
```

Comment:
weird spacing on this line making it a little hard to read (maybe due to tabs)

Reply:
Hahaha yeah, this was really annoying me too. For some reason it looks fine in the editor but weird on GitHub (I'm assuming GitHub just defaults to 4 spaces instead of the 2 we use). Will try to fix this.